snowflake-api/src/session.rs

```diff
@@ -215,10 +215,11 @@ impl Session {
             let tokens = match self.auth_type {
                 AuthType::Certificate => {
                     log::info!("Starting session with certificate authentication");
-                    #[cfg(feature = "cert-auth")]
-                    return self.create(self.cert_request_body()?).await;
```

Comment:
@colin99d this is return from the whole function itself, not the return from the match block, hence the types would not align if feature is enabled